enumerable_method.rb

```diff
@@ -0,0 +1,117 @@
+# Enumerable methods
+module Enumerable
+  def my_each
+    p enum_for(:my_each) unless block_given?
+    for value in self do
+      yield value
+    end
+    p self
+  end
+
+  def my_each_with_index
+    p enum_for(:my_each_with_index) unless block_given?
+    index = 0
+    for value in self do
+      yield(value, index)
+      index += 1
+    end
+    p self
+  end
+
+  def my_select
+    p enum_for(:my_select) unless block_given?
+    new_array = []
+    for value in self do
+      new_array.push(value) if yield value
+    end
+    p new_array
+  end
```

Comment:
Replace the 'p' with a 'return', since what the enumerables does is to return a value, also is a better practice to use always return a value and not just print in on screen